src/main/java/Lewis.java

```diff
@@ -0,0 +1,227 @@
+                        ⢣⣸⣆⡀⠀⠈⠉⠁⠀⠀⣠⣷⠈⠙⠛⠛⠛⠉⢀⣴⡊⠉⠁⠈⢢⣿⠀⠀⠀⢸⠡⠀⠁⠀⠀⠀⣠⣀⣀⣀⣀⡇⠀⢰⢁⡇⠀⠀⠀⢠
+                        ⠀⠻⣿⣟⢦⣤⡤⣤⣴⣾⡿⢃⡠⠔⠒⠉⠛⠢⣾⢿⣿⣦⡀⠀⠀⠉⠀⠀⢀⡇⢸⠀⠀⠀⠀⠀⠿⠿⠿⣿⡟⠀⢀⠇⢸⠀⠀⠀⠀⠘
+                        ⠀⠀⠈⠙⠛⠿⠿⠿⠛⠋⢰⡋⠀⠀⢠⣤⡄⠀⠈⡆⠙⢿⣿⣦⣀⠀⠀⠀⣜⠀⢸⠀⠀⠀⠀⠀⠀⠀⠀⢀⠃⠀⡸⠀⠇⠀⠀⠀⠀⠀
+                        ⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⡇⢣⠀⠀⠈⠛⠁⠀⢴⠥⡀⠀⠙⢿⡿⡆⠀⠀⢸⠀⢸⢰⠀⠀⠀⢀⣿⣶⣶⡾⠀⢀⠇⣸⠀⠀⠀⠀⠀⠀
+                        ⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠹⡀⢇⠀⠀⠀⢀⡀⠀⠀⠈⢢⠀⠀⢃⢱⠀⠀⠀⡇⢸⢸⠀⠀⠀⠈⠉⠉⠉⢱⠀⠼⣾⣿⣿⣷⣦⠴⠀⠀
+                        ⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢱⠘⡄⠀⠀⢹⣿⡇⠀⠀⠈⡆⠀⢸⠈⡇⢀⣀⣵⢨⣸⣦⣤⣤⣄⣀⣀⣀⡞⠀⣠⡞⠉⠈⠉⢣⡀⠀⠀
+                        ⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢃⠘⡄⠀⠀⠉⠀⠀⣠⣾⠁⠀⠀⣧⣿⣿⡿⠃⠸⠿⣿⣿⣿⣿⣿⣿⠟⠁⣼⣾⠀⠀⠀⠀⢠⠇⠀⠀
+                        ⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠘⡄⠹⣀⣀⣤⣶⣿⡿⠃⠀⠀⠀⠈⠁⠀⠀⠀⠀⠀⠀⠀⠀⠀⠉⠁⠀⠀⢻⣿⣷⣦⣤⣤⠎⠀⠀⠀
+                        ⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠘⣤⣿⡿⠟⠛⠉⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠈⠉⠉⠉⠀⠀⠀⠀⠀""");
+                System.exit(0);
+            }
+            case "echo" -> {
+                if (curr >= input.length()) {
+                    throw new LewisException("You have to give me something to echo!");
+                }
+                StringBuilder echo = new StringBuilder();
+                while (curr < input.length()) {
+                    echo.append(input.charAt(curr));
+                    curr++;
+                }
+
+                System.out.printf("\nLewis says: \"%s\"%n",echo);
+            }
+            case "mark" -> {
+                StringBuilder taskToMark = new StringBuilder();
+                while (curr < input.length() && input.charAt(curr) >= 48 && input.charAt(curr) <= 57) {
+                    taskToMark.append(input.charAt(curr));
+                    curr++;
+                }
+                int index;
+                try {
+                    index = Integer.parseInt(taskToMark.toString());
+                    if (index >= 1 && index <= taskList.len) {taskList.update(index, Task.Status.DONE);}
+                    else {throw new LewisException(String.format("Hey, please type a valid number between %d and %d.", 1, taskList.len));}
+                }
+                catch (NumberFormatException e) {
+                    throw new LewisException(String.format("Hey, please type a valid number between %d and %d.", 1, taskList.len));
+                }
+
+            }
+            case "unmark" -> {
+                StringBuilder taskToMark = new StringBuilder();
+                while (curr < input.length() && input.charAt(curr) >= 48 && input.charAt(curr) <= 57) {
+                    taskToMark.append(input.charAt(curr));
+                    curr++;
+                }
+                int index;
+                try {
+                    index = Integer.parseInt(taskToMark.toString());
+                    taskList.update(index, Task.Status.NOT_DONE);
+                }
+                catch (NumberFormatException e) {
+                    throw new LewisException(String.format("Hey, please type a valid number between %d and %d.", taskList.curr, taskList.end));
+                }
+
+            }
+            case "delete" -> {
+                StringBuilder taskToDelete = new StringBuilder();
```

Comment:
Variable names for task related stringBuilder does not really sound like a string, maybe you can reconsider naming them taskToDeleteToString?